src/react/hooks/useMODUKComponent/useMODUKComponent.ts

```diff
@@ -0,0 +1,38 @@
+import { useEffect, useRef } from 'react'
+
+type Component = {
+  type: 'Header'
+  element: HTMLElement
+}
+
+type ComponentNameUnion<T extends { type: string; element: HTMLElement }> = T['type']
+
+type ComponentName = ComponentNameUnion<Component>
+
+type FindElementType<C extends ComponentName> = Extract<Component, { type: C }>['element']
+
```

Comment:
This ensures that the correct ref type is returned for the element.

For example to add a Button component you would have 
```ts
// useMODUKComponent.ts
type Component = {
  type: 'Header'
  element: HTMLElement
} | 
{
  type: 'Button'
  element: HTMLButtonElement
}
```

usage
```tsx

 () => {
  const { ref: buttonRef } =useMODUKComponent('Button')
 // typeof buttonRef === HTMLButtonElement
 
   const { ref: headerRef } =useMODUKComponent('Header')
 // typeof headerRef === HTMLElement
}
```